api/rest/coordinator.go

```diff
@@ -0,0 +1,227 @@
+package rest
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+const numberOfWorkers = 10
+
+type BundleStatus struct {
+	ID   string
+	node node
+	done bool
+	err  error
+}
+
+// Coordinator is an interface to coordinate the creation of diagnostics bundles
+// across a cluster of nodes
+type Coordinator interface {
+	Create(ctx context.Context, id string, nodes []node) <-chan BundleStatus
+	Collect(ctx context.Context, bundleID string, numBundles int, statuses <-chan BundleStatus) (string, error)
+}
+
+// ParallelCoordinator implements Coordinator interface to coordinate bundle
+// creation across a cluster, parallelized across multiple goroutines
+type ParallelCoordinator struct {
+	client Client
+}
+
+// job is a function that will be called by worker. The output will be added to results chanel
+type job func() BundleStatus
+
+// worker is a function that will run incoming jobs from jobs channel and put jobs output to results chan
+func worker(_ context.Context, jobs <-chan job, results chan<- BundleStatus) {
+	for j := range jobs {
```

Comment:
From what I can see, we never close the jobs channel so these goroutines will never end.

Reply:
This gorutine should be closed when contex is canceled

Reply:
Is that something that we'll have to do ourselves or will we let the 10 minute timeout on the context handle that?

Reply:
I think waiting for timeout is fine